File: sdk/samples/Cardspace_samples/TechnologySamples/CardSpace/CreatingManagedCards/VB/CardWriter/DefaultValues.vb
Imports System
Imports System.Collections.Generic
Imports System.IdentityModel.Claims
Imports System.Text

Namespace ManagedCardWriter
    Public Class DefaultValues
        ' Methods
        Shared Sub New()
            DefaultValues.m_claims = New Dictionary(Of String, ClaimInfo)
            DefaultValues.m_claims.Add(ClaimTypes.GivenName, New ClaimInfo(ClaimTypes.GivenName, "Given Name", "Given Name"))
            DefaultValues.m_claims.Add(ClaimTypes.Surname, New ClaimInfo(ClaimTypes.Surname, "Last Name", "Last Name"))
            DefaultValues.m_claims.Add(ClaimTypes.Email, New ClaimInfo(ClaimTypes.Email, "Email Address", "Email Address"))
            DefaultValues.m_claims.Add(ClaimTypes.StreetAddress, New ClaimInfo(ClaimTypes.StreetAddress, "Street Address", "Street Address"))
            DefaultValues.m_claims.Add(ClaimTypes.Locality, New ClaimInfo(ClaimTypes.Locality, "Locality", "Locality"))
            DefaultValues.m_claims.Add(ClaimTypes.StateOrProvince, New ClaimInfo(ClaimTypes.StateOrProvince, "State or Province", "State or Province"))
            DefaultValues.m_claims.Add(ClaimTypes.PostalCode, New ClaimInfo(ClaimTypes.PostalCode, "Postal Code", "Postal Code"))
            DefaultValues.m_claims.Add(ClaimTypes.Country, New ClaimInfo(ClaimTypes.Country, "Country", "Country"))
            DefaultValues.m_claims.Add(ClaimTypes.HomePhone, New ClaimInfo(ClaimTypes.HomePhone, "Home Phone", "Home Phone"))
            DefaultValues.m_claims.Add(ClaimTypes.OtherPhone, New ClaimInfo(ClaimTypes.OtherPhone, "Other Phone", "Other Phone"))
            DefaultValues.m_claims.Add(ClaimTypes.MobilePhone, New ClaimInfo(ClaimTypes.MobilePhone, "Mobile Phone", "Mobile Phone"))
            DefaultValues.m_claims.Add(ClaimTypes.Gender, New ClaimInfo(ClaimTypes.Gender, "Gender", "Gender"))
            DefaultValues.m_claims.Add(ClaimTypes.DateOfBirth, New ClaimInfo(ClaimTypes.DateOfBirth, "Date of Birth", "Date of Birth"))
            DefaultValues.m_claims.Add(ClaimTypes.PPID, New ClaimInfo(ClaimTypes.PPID, "Site Specific ID", "Site Specific ID"))
            DefaultValues.m_claims.Add(ClaimTypes.Webpage, New ClaimInfo(ClaimTypes.Webpage, "Webpage", "Webpage"))
        End Sub


        ' Properties
        Public Shared ReadOnly Property Claims() As Dictionary(Of String, ClaimInfo)
            Get
                Return DefaultValues.m_claims
            End Get
        End Property

        Public Shared ReadOnly Property ClaimsList() As String()
            Get
                Return New String() {ClaimTypes.GivenName, ClaimTypes.Surname, ClaimTypes.Email, ClaimTypes.StreetAddress, ClaimTypes.Locality, ClaimTypes.StateOrProvince, ClaimTypes.PostalCode, ClaimTypes.Country, ClaimTypes.HomePhone, ClaimTypes.OtherPhone, ClaimTypes.MobilePhone, ClaimTypes.Gender, ClaimTypes.DateOfBirth, ClaimTypes.PPID, ClaimTypes.Webpage}
            End Get
        End Property

        Public Shared ReadOnly Property TokenTypeList() As String()
            Get
                Return New String() {"urn:oasis:names:tc:SAML:1.0:assertion", "http://docs.oasis-open.org/wss/oasis-wss-saml-token-profile-1.1#SAMLV1.1"}
            End Get
        End Property


        ' Fields
        Public Const CardId As String = "http://www.fabrikam.com/card/1"
        Public Const CardName As String = "My new card"
        Public Const CardVersion As String = "1"
        Public Const Issuer As String = "http://localhost:4444/sts"
        Public Const IssuerName As String = "Microsoft"
        Public Const Language As String = "en-us"
        Private Shared m_claims As Dictionary(Of String, ClaimInfo)
        Public Const MexUri As String = "https://localhost:4445/sts/mex"

    End Class
    ' Nested Types
    Public Enum CardType
        ' Fields
        KerberosAuth = 1
        None = 4
        SelfIssuedAuth = 2
        SmartCard = 3
        UserNamePassword = 0
    End Enum

End Namespace

File: sdk/samples/Cardspace_samples/TechnologySamples/CardSpace/CreatingManagedCards/CS/CardWriter/DefaultValues.cs
//-----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//-----------------------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.IdentityModel.Claims;
namespace ManagedCardWriter
{
    public class DefaultValues
    {
        static Dictionary<string, ClaimInfo> m_claims = new Dictionary<string, ClaimInfo>();

        public const string CardName = "My new card";
        public const string CardId = "http://www.fabrikam.com/card/1";
        public const string Issuer = "http://localhost:4444/sts";
        public const string Language = "en-us";
        public const string IssuerName = "Microsoft";
        public const string CardVersion = "1";
        public const string MexUri = "https://localhost:4445/sts/mex";

        static DefaultValues()
        {
            m_claims.Add(ClaimTypes.GivenName, new ClaimInfo(ClaimTypes.GivenName, "Given Name", "Given Name"));
            m_claims.Add(ClaimTypes.Surname, new ClaimInfo(ClaimTypes.Surname, "Last Name", "Last Name"));
            m_claims.Add(ClaimTypes.Email, new ClaimInfo(ClaimTypes.Email, "Email Address", "Email Address"));
            m_claims.Add(ClaimTypes.StreetAddress, new ClaimInfo(ClaimTypes.StreetAddress, "Street Address", "Street Address"));
            m_claims.Add(ClaimTypes.Locality, new ClaimInfo(ClaimTypes.Locality, "Locality", "Locality"));
            m_claims.Add(ClaimTypes.StateOrProvince, new ClaimInfo(ClaimTypes.StateOrProvince, "State or Province", "State or Province"));
            m_claims.Add(ClaimTypes.PostalCode, new ClaimInfo(ClaimTypes.PostalCode, "Postal Code", "Postal Code"));
            m_claims.Add(ClaimTypes.Country, new ClaimInfo(ClaimTypes.Country, "Country", "Country"));
            m_claims.Add(ClaimTypes.HomePhone, new ClaimInfo(ClaimTypes.HomePhone, "Home Phone", "Home Phone"));
            m_claims.Add(ClaimTypes.OtherPhone, new ClaimInfo(ClaimTypes.OtherPhone, "Other Phone", "Other Phone"));
            m_claims.Add(ClaimTypes.MobilePhone, new ClaimInfo(ClaimTypes.MobilePhone, "Mobile Phone", "Mobile Phone"));
            m_claims.Add(ClaimTypes.Gender, new ClaimInfo(ClaimTypes.Gender, "Gender", "Gender"));
            m_claims.Add(ClaimTypes.DateOfBirth, new ClaimInfo(ClaimTypes.DateOfBirth, "Date of Birth", "Date of Birth"));
            m_claims.Add(ClaimTypes.PPID, new ClaimInfo(ClaimTypes.PPID, "Site Specific ID", "Site Specific ID"));
            m_claims.Add(ClaimTypes.Webpage, new ClaimInfo(ClaimTypes.Webpage, "Webpage", "Webpage"));

        }

        public static Dictionary<string, ClaimInfo> Claims
        {
            get { return m_claims; }
        }

        internal static string[] ClaimsList = new string[]{
                ClaimTypes.GivenName, 
                ClaimTypes.Surname , 
                ClaimTypes.Email, 
                ClaimTypes.StreetAddress, 
                ClaimTypes.Locality, 
                ClaimTypes.StateOrProvince,
                ClaimTypes.PostalCode, 
                ClaimTypes.Country, 
                ClaimTypes.HomePhone, 
                ClaimTypes.OtherPhone,
                ClaimTypes.MobilePhone,
                ClaimTypes.Gender, 
                ClaimTypes.DateOfBirth,
                ClaimTypes.PPID, 
                ClaimTypes.Webpage};
        
        internal static string[] TokenTypeList =new string[] {
            "urn:oasis:names:tc:SAML:1.0:assertion",
            "http://docs.oasis-open.org/wss/oasis-wss-saml-token-profile-1.1#SAMLV1.1" };
        
        

        public enum CardType : int
        {
            UserNamePassword,
            KerberosAuth,
            SelfIssuedAuth,
            SmartCard,
            None

        }

    }

    public sealed class ClaimInfo
    {
        string m_displayTag;
        string m_description;
        string m_id;



        public ClaimInfo(string id, string tag, string des)
        {
            m_id = id;
            m_displayTag = tag;
            m_description = des;
        }
        public string DisplayTag
        {
            get { return m_displayTag; }
            set { m_displayTag = value; }
        }

        public string Description
        {
            get { return m_description; }
            set { m_description = value; }
        }

        public string Id
        {
            get { return m_id; }
            set { m_id = value; }
        }
    }

}